File: python/math/trend/63-dist.py
import numpy as np
import matplotlib.pyplot as plt

# Generate data points for x-axis
x = np.linspace(-5, 5, 1000)

# Calculate the corresponding y-values
# for a standard normal distribution
y = 1 / np.sqrt(2 * np.pi) * np.exp(-0.5 * x**2)

# Plot the normal distribution
plt.plot(x, y, color='blue')

# Add labels and title
plt.xlabel('x')
plt.ylabel('Density')
plt.title('Standard Normal Distribution')

# Show grid
plt.grid(True)

# Show plot
plt.show()

File: julia/trend/63-dist.jl
using StatsPlots, Distributions

# Generate data points for x-axis
x = range(-5, 5, length=1000)

# Calculate the corresponding y-values
# for a standard normal distribution
y = pdf(Normal(), x)

# Calculate the percentiles
percentiles = [25, 50, 75, 100]
quantiles = quantile.(Normal(), percentiles ./ 100)

# Plot the normal distribution
plot(
  x, y, fillrange = zero(x), fillalpha = 0.35,
  color=:black,
  label="Standard Normal Distribution", lw=1)
 
# Shade regions corresponding to percentiles
for (i, q) in enumerate(quantiles)
  sx = x .<= q
  sy = y[sx]
  # plot!(sx, sy, fillrange = zero(sx), fc=:blues)
  # vline!([q], color=i, alpha=0.3, label="")
end


# Add labels and title
xlabel!("x")
ylabel!("Density")
title!("Standard Normal Distribution with Quantiles")

# Save plot as PNG
savefig("63-quantiles.png")
